platform/avr-raven/apps/raven-lcd-interface/raven-lcd.c

```diff
@@ -146,6 +146,9 @@ raven_ping6(void)
     UIP_ICMP_BUF->icmpchksum = 0;
     UIP_ICMP_BUF->icmpchksum = ~uip_icmp6chksum();
 
+#if UIP_CONF_IPV6_RPL
+    rpl_insert_header();
+#endif /* UIP_CONF_IPV6_RPL */
 
     tcpip_ipv6_output();
```

Comment:
The more I look at it the more I think you're right, but `rpl_insert_header` might need modifying to handle all different cases.